src/main.rs

```diff
@@ -71,12 +76,12 @@ fn pair_valid(a: i32, b: i32, direction: i32) -> bool {
             } else {
                 println!("Valid: {} {} {}", a, b, direction);
             }
-            return offset != 0
+            return offset != 0;
         }
         offset += 1;
         if offset > 3 {
             println!("Not valid: {} {} {}", a, b, direction);
-            return false
+            return false;
```

Comment:
Most of these small changes are from running `rustfmt` which happens automatically in my editor.
This too is common practice with rust...